workers/data_refinery_workers/processors/illumina.py

```diff
@@ -156,6 +173,99 @@ def _detect_columns(job_context: Dict) -> Dict:
 
     return job_context
 
+def _detect_platform(job_context: Dict) -> Dict:
+    """
+    Determine the platform/database to process this sample with.
+    They often provide something like "V2" or "V 2", but we don't trust them so we detect it ourselves.
+
+    Related: https://github.com/AlexsLemonade/refinebio/issues/232
+    """
+
+    all_databases = {
+        'HOMO_SAPIENS': [
+            'illuminaHumanv1',
+            'illuminaHumanv2',
+            'illuminaHumanv3',
+            'illuminaHumanv4',
+        ],
+        'MUS_MUSCULUS': [
+            'illuminaMousev1',
+            'illuminaMousev1p1',
+            'illuminaMousev2',
+        ],
+        'RATTUS_NORVEGICUS': [
+            'illuminaRatv1'
+        ]
+    }
+
+    sample0 = job_context['samples'][0]
+    databases = all_databases[sample0.organism.name]
+
+    # Loop over all of the possible platforms and find the one with the best match.
+    highest = 0.0
+    high_mapped_percent = 0.0
+    high_db = None
+    for platform in databases:
+        try:
+            result = subprocess.check_output([
+                    "/usr/bin/Rscript", 
+                    "--vanilla", 
+                    "/home/user/data_refinery_workers/processors/detect_database.R",
+                    "--platform", platform,
+                    "--inputFile", job_context['input_file_path'],
+                    "--column", job_context['probeId'],
+                ])
+
+            results = result.decode().split('\n')
+            cleaned_result = float(results[0].strip())
+
+            if cleaned_result > highest:
+                highest = cleaned_result
+                high_db = platform
+                high_mapped_percent = float(results[1].strip())
+
+        except Exception as e:
+            logger.exception(e)
+            continue
+
+    # If the match is over 75%, record this and process it on that platform.
+    if high_mapped_percent > 75.0:
+        job_context['platform'] = high_db
+
+        for sample in job_context['samples']:
+            sa = SampleAnnotation()
+            sa.sample = sample
+            sa.data = {
+                "detected_platform": high_db, 
+                "detection_percentage": highest,
+                "mapped_percentage": high_mapped_percent
```

Comment:
These seem like some useful statistics to have around even if we're just going to NO_OP the sample. I could see us wanting to look into the stats of all the samples we weren't confidant enough to process at some point.

Reply:
I agree.